application/backend/src/business/services/jobs/job-copy-out-service.ts

```diff
@@ -0,0 +1,344 @@
+      }
+
+      await e
+        .insert(e.job.CopyOutJob, {
+          forRelease: releaseQuery,
+          status: e.job.JobStatus.running,
+          started: e.datetime_current(),
+          percentDone: e.int16(0),
+          messages: e.literal(e.array(e.str), ["Started copy out in AWS"]),
+          // we save a pointer to our related audit entry so that we can close it off later
+          auditEntry: e
+            .select(e.audit.ReleaseAuditEvent, (ae) => ({
+              filter: e.op(ae.id, "=", e.uuid(newAuditEventId)),
+            }))
+            .assert_single(),
+          awsExecutionArn: startExecutionResult.executionArn,
+        })
+        .run(tx);
+    });
+
+    // return the status of the release - which now has a runningJob
+    return await this.releaseService.getBase(releaseKey, userRole);
+  }
+
+  /**
+   * Progress the work of the copy out job. As it turns out,
+   * the work just involves checking the status of the AWS steps
+   * function we invoked.
+   *
+   * @param jobId
+   */
+  public async progressCopyOutJob(jobId: string): Promise<number> {
+    // TODO some security level here? does the user have permissions?
+    //      this method is only ever called by the job handler which acts with system level permissions??
+
+    return await this.edgeDbClient.transaction(async (tx) => {
+      const copyOutJob = await this.getCurrentJobWithExceptionForInvalid(
+        tx,
+        jobId
+      );
+
+      // get the status of the overall steps function that does copying
+      const describeExecutionResult = await this.sfnClient.send(
+        new DescribeExecutionCommand({
+          executionArn: copyOutJob.awsExecutionArn,
+        })
+      );
+
+      assert(
+        describeExecutionResult,
+        `Steps execution ARN ${copyOutJob.awsExecutionArn} unexpectedly disappeared`
+      );
+
+      // at our first polling after a successful launch - we upgrade to 1%
+      if (copyOutJob.percentDone === 0)
+        await this.updateCurrentJob(tx, jobId, {
+          percentDone: 1,
+        });
+
+      // now we try to look into the map run (the actual copy) to see if we can get some stats from it
+      {
+        const listMapRunsResults = await this.sfnClient.send(
+          new ListMapRunsCommand({
+            executionArn: copyOutJob.awsExecutionArn,
+          })
+        );
+
+        // we have only one map run in our steps so this can only ever be 1 or 0
+        if (
+          listMapRunsResults.mapRuns &&
+          listMapRunsResults.mapRuns.length === 1
+        ) {
+          const describeMapResult = await this.sfnClient.send(
+            new DescribeMapRunCommand({
+              mapRunArn: listMapRunsResults.mapRuns[0].mapRunArn,
+            })
+          );
+
+          // if we have an item count report out the percent done
+          if (describeMapResult.itemCounts) {
+            const countDone =
+              describeMapResult.itemCounts.failed! +
```

Comment:
Unless I misunderstand, I think it should be easy to avoid disabling the type checker here:

```suggestion
              (describeMapResult.itemCounts?.failed + 0) +
```

etc

Reply:
The "failed", "total" (etc) properties are required as per the AWS documentation - but for some reason that isn't reflected in the type of itemCount. 

https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-sfn/interfaces/maprunitemcounts.html
vs
https://docs.aws.amazon.com/step-functions/latest/apireference/API_MapRunItemCounts.html

(the ! doesn't disable the type checker - it makes a non-null assertion - the type is then still retained as a number)